pkg/publish/kind/write.go

```diff
@@ -53,9 +54,15 @@ func Tag(src, dest name.Tag) error {
 	})
 }
 
+// writeMu serializes image imports to kind, to avoid locking situations
+// observed when publishing multiple images to KinD at the same time.
+var writeMu sync.Mutex
```

Comment:
This works but package-level mutex is kinda gross.

We could do this here, given that we usually have one publisher that we hold (I think): https://github.com/google/ko/blob/75ab99123d2c6b830fc11c935f7d3a2a2d740256/pkg/publish/kind.go#L105